### ISSUE TITLE
# 损失函数总结

### ISSUE THREAD

**Issue by huuuuusy** [huuuuusy/Deep-Learning-Practice-Everyday#3]

**Comment by huuuuusy** [comment]
# 均值平方差(MSE)

MSE指参数估计值和参数真实值之间的差的平方的期望值。
- 公式:
![image](https://user-images.githubusercontent.com/32117906/63755418-2f6b3100-c8e9-11e9-9631-0690b6d2928d.png)

**Comment by huuuuusy** [comment]
# 交叉熵(Cross Entropy)

交叉熵用于预测输入样本属于某一类的概率。
- 公式：
![image](https://user-images.githubusercontent.com/32117906/63756800-8ffb6d80-c8eb-11e9-9f5a-0b97be90dde5.png)
其中a代表预测值，y代表ground truth(0或者1)
- 注意：
    - a经过分布统一化处理(或者经过Sigmoid函数激活),取值在(0,1)之间
    - 如果真实值和预测值均为1,则loss为0;反之loss为其他数值
- TF指令：
    - Sigmoid交叉熵：
        - tf.nn.sigmoid_cross_entropy_with_logits(logits, targets, name = None)
        - 计算预测值logits和真实值targets之间的交叉熵
    - Softmax交叉熵：
        - tf.nn.softmax_cross_entropy_with_logits(logits, labels, name = None)
        - 计算预测值logits和真实值labels之间的交叉熵
        - logits和labels必须为相同的shape和数据类型

**Comment by huuuuusy** [comment]
# 小结

损失函数的选取取决于输入数据的类型：
- 如果输入的是实数、无界的值，损失函数使用MSE
- 如果输入的是位矢量(分类标志)，使用交叉熵更合适
